src/util.rs

```diff
@@ -55,7 +55,9 @@ impl Fairing for AppHeaders {
         res.set_raw_header("Referrer-Policy", "same-origin");
         res.set_raw_header("X-Content-Type-Options", "nosniff");
         res.set_raw_header("X-Robots-Tag", "noindex, nofollow");
-        res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+        if !req_uri_path.ends_with("/icon.png") {
+            res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
+        }
```

Comment:
Isn't something like that also needed for "mail-github.png" and "logo-gray.png" in case `SMTP_EMBED_IMAGES=false` (so they can be fetched by mail clients)? Something like
```suggestion
        if !(req_uri_path.ends_with("/icon.png")
             || (!CONFIG.smtp_embed_images
                && (req_uri_path.ends_with("/logo-gray.png")
                || req_uri_path.ends_with("/mail-github.png"))) {
            res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
        }
```

Reply:
Hmmm, i have not tested that yet to be fair.
But then only excluding those files will probably still cause issues, since you could server custom image files too, not just those we set our self.

Reply:
I think there is a better way.

Reply:
After checking the email I received from Vaultwarden, I found that the two files are under `/vw_static/`.
How about doing it like this?

```Rust
if let Some(path_stripped) = req_uri_path.strip_prefix(&*CONFIG.domain_path()) {
    if !(path_stripped.starts_with("/icons/") || path_stripped.starts_with("/vw_static/") {
        res.set_raw_header("Cross-Origin-Resource-Policy", "same-origin");
    }
}
```

Reply:
Sounds good. I mean, if we want to tackle this in a separate PR this would be fine for me too.

Reply:
We can use `res.content_type()` and check if it is an image :)

Reply:
Isn't it too much to not apply `Cross-Origin-Resource-Policy` to all images?

Reply:
I think we could use `res.headers().get_one("Content-Type").is_some_and(|v| v.starts_with("image"))` which should solve it.

Well, what if someone uses a modified image from the web-vault somewhere? Like browsers might use it for the favicon of the domain for example?

It's not that the images really need protection from being fetched in my opinion.
But more to prevent other scripts or sites from being able to abuse the API endpoints.